### PR TITLE
Define exceptions in HackAndSlash

### PR DESCRIPTION
**Pull request by ipdae**
- Use defined exceptions instead of  `InvalidOperationException`

### CODE REVIEW

**Review by longfin — APPROVED** [review]

File: Lib9c/Model/WorldInformation.cs [hunk 1]
```diff
@@ -240,24 +240,16 @@ public bool TryAddWorld(WorldSheet.Row worldRow, out World world)
             return true;
         }
 
-        public bool TryUpdateWorld(WorldSheet.Row worldRow, out World world)
+        public void UpdateWorld(WorldSheet.Row worldRow)
         {
-            if (worldRow is null ||
```
Comment (boscohyun) [inline Lib9c/Model/WorldInformation.cs:245]: Why did you remove the `null` check?

File: Lib9c/Model/WorldInformation.cs [hunk 2]
```diff
@@ -397,6 +389,21 @@ public bool TryGetLastClearedStageId(out int stageId)
...
 
+        public void AddNewWorld(WorldSheet.Row worldRow, long unlockedAt, WorldSheet worldSheet)
+        {
+            var worldId = worldRow.Id;
+            if (IsStageCleared(worldRow.StageBegin - 1))
+            {
+                var world = new World(worldRow);
+                _worlds.Add(worldId, world);
+                UnlockWorld(worldId, unlockedAt, worldSheet);
+            }
```
Comment (boscohyun) [inline Lib9c/Model/WorldInformation.cs:400]: If it was to unlock while adding `world`, it would have been better to have a detailed name like `AddNewWorldAndUnlock()`. However, I can guess because there is `unlockedAt` in the argument.

**Review by boscohyun — APPROVED** [review]

File: Lib9c/Model/WorldInformation.cs (hunk 1 above)
Comment (ipdae) [reply Lib9c/Model/WorldInformation.cs:245]: `UpdateWorld` called only in `HackAndSlash`, and null check before call `UpdateWorld`.
